contracts/erc721/UserMintableERC721.sol

```diff
@@ -430,6 +464,65 @@ contract UserMintableERC721 is Initializable, ERC721, ERC721Enumerable, ERC721Me
     );
   }
 
+  /**
+   * @dev Checks that provided document is signed by an identity
+   * and validates and checks if the public key used is a valid
+   * SIGNING_KEY
+   * @param documentRoot bytes32 the anchored document root
+   * @param identity address Identity that signed the document
+   * @param signingRoot bytes32 The message that was signed
+   * @param signature bytes The signature
+   * used to contract the property for precise proofs
+   * @param salt bytes32 salt for leaf construction
+   * @param proof bytes32[] proofs for leaf construction
+   */
+
+  function _requireSignedByIdentity(
+    bytes32 documentRoot,
+    address identity,
+    bytes32 signingRoot,
+    bytes memory signature,
+    bytes32 salt,
+    bytes32[] memory proof
+  )
+  internal
+  view
+  {
+
+    // Extract the public key from the signature
+    bytes32 pbKey_ = bytes32(
```

Comment:
I think for external readers this always confusing.

In Ethereum the address are the first 20 bytes of the public key hash  `sha256(publicKey))`

We getting the address from `recover` method  and casting it to byte32 and calling it key.